java/connector-node/risingwave-sink-mock-flink/risingwave-sink-mock-flink-runtime/src/main/java/com/risingwave/mock/flink/runtime/FlinkDynamicAdapterFactory.java

```diff
@@ -0,0 +1,141 @@
+        Set<ConfigOption<?>> configOptions = new HashSet<>();
+        configOptions.addAll(dynamicTableSinkFactory.requiredOptions());
+        configOptions.addAll(dynamicTableSinkFactory.optionalOptions());
+        config.processOption(configOptions);
+
+        Schema.Builder schemaBuilder = Schema.newBuilder();
+        flinkColumns.forEach(
+                (column) -> schemaBuilder.column(column.getName(), column.getDataType()));
+        Schema shcema = schemaBuilder.primaryKey(tableSchema.getPrimaryKeys()).build();
+
+        ResolvedCatalogTable resolvedCatalogTable =
+                new ResolvedCatalogTable(
+                        CatalogTable.of(
+                                shcema, null, tableSchema.getPrimaryKeys(), config.getOption()),
+                        ResolvedSchema.of(flinkColumns));
+
+        FactoryUtil.DefaultDynamicTableContext defaultDynamicTableContext =
+                new FactoryUtil.DefaultDynamicTableContext(
+                        objectIdentifier,
+                        resolvedCatalogTable,
+                        new HashMap<String, String>(),
+                        new Configuration(),
+                        contextClassLoader,
+                        false);
+        DynamicTableSink.Context context = new DynamicTableSinkContextImpl();
+        // Start with the default value, and add as needed later
+        return dynamicTableSinkFactory
+                .createDynamicTableSink(defaultDynamicTableContext)
+                .getSinkRuntimeProvider(context);
+    }
+
+    /**
+     * According to different implementations, create different sink writer, and package it as a
+     * RW-recognizable method(SinkWriterV1), after which the RW can call the method to complete
+     * write directly.
+     */
+    @Override
+    public SinkWriter createWriter(TableSchema tableSchema, Map<String, String> tableProperties) {
+        ObjectMapper mapper = new ObjectMapper();
+        FlinkDynamicAdapterConfig flinkDynamicAdapterConfig =
+                mapper.convertValue(tableProperties, FlinkDynamicAdapterConfig.class);
+
+        DynamicTableSink.SinkRuntimeProvider sinkRuntimeProvider =
+                buildDynamicTableSinkProvider(tableSchema, flinkDynamicAdapterConfig);
+        if (sinkRuntimeProvider instanceof SinkProvider) {
+            Sink<RowData, ?, ?, ?> sink = ((SinkProvider) sinkRuntimeProvider).createSink();
+            return new SinkWriterV1.Adapter(new SinkWriterImpl<>(tableSchema, sink));
+        } else if (sinkRuntimeProvider instanceof SinkV2Provider) {
+            org.apache.flink.api.connector.sink2.Sink<RowData> sink =
+                    ((SinkV2Provider) sinkRuntimeProvider).createSink();
+
+            try {
+                if (sink.createWriter(new SinkWriterContextV2()) instanceof AsyncSinkWriter) {
+                    return new SinkWriterV1.Adapter(new AsyncSinkWriterImpl(tableSchema, sink));
```

Comment:
`SinkWriterV1` is already deprecated. We should implement the `SinkWriter` for the adapter.

Reply:
okk!  I noticed that we have other sinks that implement SinkWriterV1, should we migrate them all, or is there any reason here why we must implement SinkWriterV1?

Reply:
We just wanted to avoid changing the old implementation on v1 when we introduced the new `SinkWriter` interface. 

We can keep the current implementation untouched before we implement new feature on any of them, and then we can do the refactor on the sink by the way.